### ISSUE TITLE
[FEAT] Have an optional to reset from when close modal inside `useModalForm`

### ISSUE THREAD

**Issue by Dominic-Preap** [refinedev/refine#5954]
### Is your feature request related to a problem? Please describe.

Hi, I would like to have option to reset the form value to `defaultValues` when use `close()` in `useModalForm` hooks.

There are 2 issues, I'm facing:

- When using create `useModalForm({refineCoreProps:{action: 'create'})` I want to reset the form when user just `close()` (cancel by chance) and when user open modal again, the previous value is still there. 
- When using edit `useModalForm({refineCoreProps:{action: 'edit'})`, when close modal and open another modal by call `show(id)`, the previous data is still inside the new modal and waiting to replace after fetch is completed. This will make user experience confuse sometimes when the internet is slow. 

To bypass the issue, we call `reset()` then `close()`.

### Describe alternatives you've considered

It would be nice to have another option inside `useModalForm` when user want to reset form after closing the form not just by summitted. For example 

```ts
useModalForm({
    refineCoreProps: { action: 'create' },
    defaultValues,
    modalProps: {
      autoResetFormWhenClose: true // <-- something like this
    }
    resolver: valibotResolver(schema)
  });
```

### Additional context

N/A

### Describe the thing to improve

N/A

**Comment by rakshixh** [comment]
Hey @Dominic-Preap and @alicanerdurmaz ,

I would like to work on this issue. Can you assign it to me? But please note that I do need to take some time to understand the code base and to go through the contributing guidelines. Thanks

**Comment by GANES1998** [comment]
@Dominic-Preap and @alicanerdurmaz, I am interested in this issue. Can you please assign it to me? Also, can you please guide me on the solution

**Comment by Xajid** [comment]
@Dominic-Preap I'd like to work on this one. Please assign it to me, I'd be happy to contribute.

**Comment by BatuhanW** [comment]
Hey @GANES1998 assigned issue to you, if we don't hear back from you, we'll assign it to @Xajid

**Comment by GANES1998** [comment]
Hi @BatuhanW, Thanks for assigning to me. I'll work on this.

**Comment by bhargavpshah98** [comment]
Hey @Dominic-Preap , I am interested to work on this issue. Can you please assign it to me if it is still unsolved ?

**Comment by BatuhanW** [comment]
@bhargavpshah98 are you still interested working on this issue?

**Comment by bhargavpshah98** [comment]
Hello @BatuhanW , Yes I am interested

**Comment by abdulqdaer-q** [comment]
Hey @Dominic-Preap  and @alicanerdurmaz , if this is still available can u assign it to me ?

**Comment by BatuhanW** [comment]
Hey @bhargavpshah98, it seems the notification lost and didn't see your comment. Just assigned to you, looking forward to your PR!

@abdulqdaer-q since @bhargavpshah98 asked earlier, I assigned to him. We can assign to you if he won't work on this one.

**Comment by saurabh-619** [comment]
hey @BatuhanW, is this issue still open? if so please assign it to me, I'd like to contribute.

**Comment by paoloLigsay** [comment]
Hi @BatuhanW, I can work on this issue, can you assign this to me if it's still unresolved?

**Comment by aliemir** [comment]
Hey everyone! I had some time to break down the issues here.

For the `action: "create"` persisting the dirty values; I think `useModalForm({ resetOnClose: boolean })` can be added for this behavior 🤔

I hope understood correctly @Dominic-Preap, when calling `close(); show("1");` modal is kept open and values are reset to the previous value for a moment (until the next fetch is complete). I think when setting the `initialValues` in `useModalForm().formProps` we currently do:

https://github.com/refinedev/refine/blob/013e975f15087561435b9e31c813b9d119d19d39/packages/antd/src/hooks/form/useForm.ts#L318

By default, `queryOptions.keepPreviousData` is set to `true` by Refine. By changing this to `false`, I was able to clear the previous values while fetching the new record.  🚀 

@paoloLigsay Can you also check if setting `keepPreviousData: false` helps in your case as well? 🙏 

I think the change can be done here in `useForm` hook of `@refinedev/antd` and work for `useModalForm` too.

https://github.com/refinedev/refine/blob/013e975f15087561435b9e31c813b9d119d19d39/packages/antd/src/hooks/form/useForm.ts#L256

If this resolves the issue, we're open to contributions for the fix 🙏

**Comment by ahmedmelfay** [comment]
```autoSubmitClose``` doesn't appear to be wroking either

**Comment by aqir-sapaad** [comment]
@BatuhanW 
 can you please assign this to me i will love to contribute .

**Comment by BatuhanW** [comment]
Hey @aqir-sapaad assigned to you.

**Comment by aqir-sapaad** [comment]
sorry it was my work email can you please assign to me here @BatuhanW

**Comment by KGMaxey** [comment]
Ran into this issue myself. I did a little playing around and if I use a `create` action but still set an id, for example and empty string `""`, the reset on cancel works.

I traced that id around in code for a bit and it appears that this happens because the `resetFields` is called in this `useEffect`:
https://github.com/refinedev/refine/blob/26d4065117001290a6f8d94f3f78c37e213b8947/packages/antd/src/hooks/form/useForm.ts#L275

I think this useEffect is triggered when `setId(undefined)` is called in the `handleClose` of `useModalForm` here:
https://github.com/refinedev/refine/blob/26d4065117001290a6f8d94f3f78c37e213b8947/packages/antd/src/hooks/form/useModalForm/useModalForm.ts#L292

When the `action` is set to `edit` you have to specify an `id` so setting it to `undefined` triggers the `useEffect`. But on create when you don't set an `id` it's already `undefined` and doesn't trigger.

**Comment by KGMaxey** [comment]
Doing this does appear to have the problem of `defaultFormValues` not being set if I then re-open the create modal though

**Comment by ChevrierDev** [comment]
Hey, @Dominic-Preap  I'd want to contribute to this issue  🚀

**Comment by BatuhanW** [comment]
Hey @ChevrierDev I've assigned this issue to you.

**Comment by armanrozika** [comment]
just faced with this issues. Tried to edit and save, and then open the same item, the form is empty. But if i open other items, the value is there again. It seems like when i open the same item (this is probably reading from id), the value didn't get inserted into the form. 
And, I tried to edit some value and then cancel; then when i open the same item again, the value is the value i cancelled before, not value from data fetching

**Comment by abkr08** [comment]
Hello @BatuhanW, could you assign this issue to me? I'd love to fix it once and for all.

**Comment by armanrozika** [comment]
I somehow solved this problem by setting cacheTime:0 and structuralFileSharing:false in the queryOptions

**Comment by imHardik1606** [comment]
Hi there.

If this issue is still unsolved. I would like to work on this feature to add an optional state reset when closing the modal inside useModalForm. Could you please assign this issue to me?

Thank you!

**Comment by BatuhanW** [comment]
> Hello @BatuhanW, could you assign this issue to me? I'd love to fix it once and for all.

Assigned issue to you.

@imHardik1606 if we don't hear back from @abkr08, we'll assign it to you.

**Comment by abkr08** [comment]
Thanks, @BatuhanW. I will get it sorted.

**Comment by Ananmayd** [comment]
@Dominic-Preap I would love to contribute to this issue.

**Comment by BatuhanW** [comment]
@Ananmayd assigned issue to you. 

Please make sure to thoroughly read and follow our contribution guide. PRs that doesn't follow our contribution guide, will be closed. Thanks.

https://refine.dev/docs/guides-concepts/contributing/

**Comment by erharsh237** [comment]
Hey @Dominic-Preap, I came through this issue. Can i work on it, if it is still unsolved?

**Comment by BatuhanW** [comment]
@erharsh237 assigned to you.

**Comment by erharsh237** [comment]
Hi @BatuhanW, Can you please guide me on solving this issue.

**Comment by erharsh237** [comment]
Hi @BatuhanW, I have solved the issue and all the tests have been passed. but when i am trying to push it to github, it is saying "remote: Permission to refinedev/refine.git denied to erharsh237.
fatal: unable to access 'https://github.com/refinedev/refine.git/': The requested URL returned error: 403". Can you please help me with this?

**Comment by mohit-2003** [comment]
> Hi [@BatuhanW](https://github.com/BatuhanW), I have solved the issue and all the tests have been passed. but when i am trying to push it to github, it is saying "remote: Permission to refinedev/refine.git denied to erharsh237. fatal: unable to access 'https://github.com/refinedev/refine.git/': The requested URL returned error: 403". Can you please help me with this?

It looks like you haven't fork the repo, trying to push in main one.

**Comment by ashishk15678** [comment]
is it closed ? cuz i have the changes ready , so if i can push.

**Comment by Diwakarji701** [comment]
useModalForm({
    refineCoreProps: { action: 'create' },
    defaultValues,
    modalProps: {
        autoResetFormWhenClose: true // Automatically reset form when modal is closed
    },
    resolver: valibotResolver(schema)
})

**Comment by AbiodunVlad** [comment]
I build with React on a daily basis and I want to fix this issue because I especially work with forms.

**Comment by Shauryaditya** [comment]
If this issue is still available I would like to contribute

**Comment by arunaabhs** [comment]
Is the issue still there? If yes. then please assign it to me as I want to contribute to it.

**Comment by BatuhanW** [comment]
Hello folks, you can create a PR for the issue. We aren't assigning issues anymore, sometimes assignees disappears and other candidates blocked by them.

**Comment by sobiya-22** [comment]
Hey @BatuhanW,
I was going through this issue and just wanted to confirm — do we need a separate reset button for the modal, or is the issue about this:
When a user fills out the form but closes the modal without saving, and then clicks "Create" again, the form still shows the previously typed data instead of starting fresh?
